framework/src/modules/chain/logic/dapp.js

```diff
@@ -37,19 +36,24 @@ __private.unconfirmedAscii = {};
  * @requires bytebuffer
  * @requires valid
  * @requires helpers/dapp_categories
- * @param {Storage} storage
- * @param {Object} logger
- * @param {ZSchema} schema
- * @param {Object} network
+ * @param {Object} dependencies
+ * @param {Object} dependencies.components
+ * @param {Object} dependencies.libraries
+ * @param {Storage} dependencies.components.storage
+ * @param {logger} dependencies.components.logger
+ * @param {ZSchema} dependencies.libraries.schema
+ * @param {Object} dependencies,libraries.network
  * @todo Add description for the params
  */
 class DApp {
-	constructor(storage, logger, schema, network) {
-		library = {
-			storage,
-			logger,
-			schema,
-			network,
+	constructor({ components, libraries }) {
```

Comment:
In upcoming PR I had refactored the top level scope, you can look at here. 

https://github.com/LiskHQ/lisk/blob/7c89e1a5ed7e06e9f8c70566d1eaca471f972519/framework/src/modules/chain/chain.js#L106-L128

Would be nice if all dependencies passing to these logic constructors have same naming and structure of the top level scope.

Reply:
As I have just discussed with Nazar, I think we need to discuss and seal the structure of `scope` to make it consistent across logic files. @nazarhussain , @diego-G @MaciejBaj

Reply:
This is how the scope structure has been set after discussion:
```
const scope = { 
  // static objects
  lastCommit, 
  build,
  config,
  genesisBlock,

  // This three will be out to other framework modules
  webSocket, 
  network,
  swagger,

  schema,
  bus,
  ed,
  sequence,
  balancesSequence,
  
  components: { 
    storage, 
    cache, 
    logger,
  },
  logic: {}, // Logic files
  modules: {}, // Modules
};
```